pkg/security/security.go

```diff
@@ -231,3 +243,8 @@ type SecretItem struct {
 
 	ExpireTime time.Time
 }
+
```

Comment:
Do we need both TokenExchangeer and TokenFetcher ? 
Can we just use a single "TokenGetter" interface -  with multiple implementations, selected by config ? 

The TokenGetter can do any platform-specific exchange.

Reply:
Token exchanger is for talking to Google service. CredentialFetcher is to fetch platform specific credential, which can work with any CA. They are completely different.

Reply:
My request is to use the GRPC interface and implementation. That allows abstracting the google-specific exchange functionality - without exposing it outside. 

Istio agent should just configure the GRPC per-call option, and the implementation of that can do anything that google needs (in TokenExchanger), preferably in a gRPC-maintained implementation so they can have feature parity with Istio.

But if we go with the interface I'm proposing ( gRPC one ) - TokenExchanger can also be removed, it's not 'completely different' since both are part of getting a token, just badly abstracted.